lib/cocoapods-core/podfile/target_definition.rb

```diff
@@ -296,6 +296,14 @@ def script_phases
 
       #--------------------------------------#
 
+      # @return [String] The project name to use for the given pod name or `nil` if none specified.
+      #
+      def project_name_for_pod(pod_name)
+        raw_project_names_hash[pod_name]
```

Comment:
should this use `Specification.root_name`? since the name is stored based on the root

Reply:
Very good point I will add a test .

Reply:
well did you mean for storing or querying?

Reply:
There is no spec being passed here and below we _always_ store the `pod_name = Specification.root_name(name)` into the hash.

So I think this is fine?

Reply:
well if this is called like `project_name_for_pod('SomePod/Subspec')` it would always return nil

Reply:
if that's intended then some docs for the argument could clear that up

Reply:
agreed. I will update docs.